### ISSUE TITLE
Potential bug/leak: middleware cache grows even if no middleware created

### ISSUE THREAD

**Issue by BrainCrumbz** [videojs/video.js#8653]
### Description

TL;DR: when videojs players are instantiated and disposed repeatedly on page, `middlewareInstances` keeps growing with new keys and null values.

We are hunting potential memory leaks in our video-based app. Comparing heap snapshots on Windows Chrome and Android Chrome, we noticed a (small?) amount of property arrays due to `middlewareInstances`. Snapshot were taken after disposing a player once, and then after instantiating/disposing e.g. 10 times. 

Adding logpoints in `tech/middleware.js`, we noticed that:
1. [`getOrCreateFactory`](https://github.com/videojs/video.js/blob/2f1cc6c2b38182bfb9a4dd77a2a27cd8dd38687c/src/js/tech/middleware.js#L257) is never called, so no new `middlewareInstances` is set for a new player
2. [`clearCacheForPlayer`](https://github.com/videojs/video.js/blob/2f1cc6c2b38182bfb9a4dd77a2a27cd8dd38687c/src/js/tech/middleware.js#L246) is called every time player is disposed, so a new null value is add to `middlewareInstances`

This happened for videojs v7.20.3. Comparing file with [latest revision](https://github.com/videojs/video.js/compare/v7.20.3...main#diff-166dcdb09385d794b7e633d8b85f67ef61f04f0bb4bf3fc7367fbf37025a66bb) showed no significant change, so maybe the behaviour is the same in latest code.

If everything is correct, would you consider a tiny PR that addresses this? E.g. `clearCacheForPlayer' could check if key exists in the first place?

Thanks

### Reduced test case

None. Willing to create one if issue is of interest

### Steps to reproduce

1. After page load, create a player with `videoPlayer = videojs(videoElem, videoPlayerOptions);` and set a source on it with `videoPlayer.src({src: ..., type: ...})`
2. After a timeout, dispose player with `videoPlayer.dispose();`
3. After a timeout, repeat the process e.g. twice

### Errors

None

### What version of Video.js are you using?

7.20.3

### Video.js plugins used.

None

### What browser(s) including version(s) does this occur with?

Chrome 123

### What OS(es) and version(s) does this occur with?

Windows 11

**Comment by mister-ben** [comment]
A PR would be welcome, thanks.

**Comment by BrainCrumbz** [comment]
PR can be found at https://github.com/videojs/video.js/pull/8674

Waiting for comments etc. Bye!
